### PR TITLE
SUBMARINE-1182. Use the unique source for model management and use it for later serving

### PR DESCRIPTION
**Pull request by KUAN-HSUN-LI**
### What is this PR for?
1. Use the unique source of the model to make the following serving work
2. Provide create model version REST API
3. Fix bugs

### What type of PR is it?
[Bug Fix | Feature]

### Todos
* [x] - Provide unique model source.
* [x] - Serving a model created by experiment.
* [ ] - Update document.
* [ ] - Provide tests.

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-1182

### Screenshots (if appropriate)
1. PyTorch example
https://user-images.githubusercontent.com/38066413/151348577-6a6c14a3-3404-43e7-92b1-f0a7a63bb6fe.mp4
2. TensorFlow example
https://user-images.githubusercontent.com/38066413/151350155-669d8442-63f3-4349-90f9-66c2f6ae6c35.mp4

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? Yes

### CODE REVIEW

**Comment by KUAN-HSUN-LI** [comment]
@pingsutw  @jeff-901 Help me review the PR. Thanks!

File: submarine-sdk/pysubmarine/submarine/artifacts/repository.py [hunk 1]
```diff
@@ -19,45 +19,36 @@
...
+            Prefix=f"{dest_path}/",
             Delimiter="/",
         )
         return response.get("CommonPrefixes")
 
-    def log_artifact(self, local_file: str, artifact_path: str) -> None:
-        dest_path = self.dest_path
-        dest_path = os.path.join(dest_path, artifact_path)
+    def log_artifact(self, dest_path: str, local_file: str) -> None:
+        dest_path = os.path.join(dest_path)
```
Comment (pingsutw) [inline submarine-sdk/pysubmarine/submarine/artifacts/repository.py:43]: we didn't join any other path here, could we remove this line?

**Review by pingsutw — COMMENTED** [review]
one question in the comment, others LGTM

File: submarine-sdk/pysubmarine/submarine/artifacts/repository.py (hunk 1 above)
Comment (KUAN-HSUN-LI) [reply submarine-sdk/pysubmarine/submarine/artifacts/repository.py]: Sure

**Review by pingsutw — APPROVED** [review]

**Pull request closed without merging**